src/js/background/sync.js

```diff
@@ -343,6 +343,14 @@ async function reconcileIdentities(){
     const deletedCookieStoreId = 
       await identityState.lookupCookieStoreId(deletedUUID);
     if (deletedCookieStoreId){
+      if (excludeRegExp) {
+        const deletedIdentity = await identityState.get(deletedCookieStoreId);
+        // skip excluded identities
+        if (excludeRegExpMatchesIdentity(excludeRegExp, deletedIdentity)) {
+          continue;
+        }      
+      }
```

Comment:
Any chance we can unnest this a bit and assign the last if condition to a variable. Should be easier to read and refactor in the future if we need to check for other things.